namui/src/namui/render/text_input/get_selection_on_mouse_down.rs

```diff
@@ -0,0 +1,98 @@
+use super::Props;
+use crate::namui::{self, get_text_width_internal, managers, TextInput};
+use std::ops::Range;
+
+impl TextInput {
+    pub(crate) fn get_selection_on_mouse_movement(
+        &self,
+        props: &Props,
+        click_x: f32,
+        is_dragging_by_mouse: bool,
+    ) -> Option<Range<usize>> {
+        let (font, is_shift_key_pressed) = {
+            let managers = managers();
+            let font = managers.font_manager.get_font(&props.text_param.font_type);
+
+            let is_shift_key_pressed = managers
+                .keyboard_manager
+                .any_code_press(&[namui::Code::ShiftLeft, namui::Code::ShiftRight]);
+
+            (font, is_shift_key_pressed)
+        };
+        if font.is_none() {
+            return None;
+        };
+        let font = font.unwrap();
+
+        // const continouslyFastClickCount: number;
+
+        // if (continouslyFastClickCount >= 3) {
+        //   return getMoreTripleClickSelection({ text });
+        // }
+        // if (continouslyFastClickCount === 2) {
+        //   return getDoubleClickSelection({ text, font, x: localX });
+        // }
```

Comment:
What is this comment for?

Reply:
Aaah, I believe this will be used in the future

Reply:
that's right, the code from past typescript code.